WordPress/Classes/Utility/Media/ImageDownloader.swift

```diff
@@ -103,6 +105,35 @@ actor ImageDownloader {
     // MARK: - Networking
 
     private func data(for request: URLRequest, options: ImageRequestOptions) async throws -> Data {
+        let requestKey = request.urlRequest?.url?.absoluteString ?? ""
+        let task = tasks[requestKey] ?? ImageDataTask(task: Task {
+            try await self._data(for: request, options: options, key: requestKey)
+        })
+        let subscriptionID = UUID()
+        task.subscriptions.insert(subscriptionID)
+        tasks[requestKey] = task
+
+        return try await withTaskCancellationHandler {
+            try await task.task.value
+        } onCancel: {
+            Task {
+                await self.unsubscribe(subscriptionID, key: requestKey)
```

Comment:
[Q] Why is unsubscribe(_: key:) async without explicitly being marked as async?

Reply:
Oh - is it because Task { ... } isn't isolated to ImageDownloader?

Reply:
The `onCancel` closure is not isolated to `ImageDownloader`, but the `unsubscribe` needs to be actor-isolated on `ImageDownloader`. The only way to call an actor-isolated method from "outside" is from an async function. It's a bit more complicated than I maybe wanted it to be, but it's still better than the traditional approach of wrapping every method in `queue.async` / GCD.